src/comment_provider/conversationProvider.ts

```diff
@@ -0,0 +1,575 @@
+import {
+  CancellationToken,
+  Comment,
+  CommentAuthorInformation,
+  CommentMode,
+  CommentReply,
+  CommentThread,
+  CommentThreadState,
+  Disposable,
+  MarkdownString,
+  Range,
+  TextDocument,
+  Uri,
+  comments,
+  env,
+  window,
+  workspace,
+} from "vscode";
+import { extendErrorWithSupportLinks, provideSingleton } from "../utils";
+import { DBTTerminal } from "../dbt_client/dbtTerminal";
+import path = require("path");
+import {
+  Conversation,
+  ConversationGroup,
+  ConversationService,
+  SharedDoc,
+} from "../services/conversationService";
+import { SharedStateService } from "../services/sharedStateService";
+import { UsersService } from "../services/usersService";
+import { QueryManifestService } from "../services/queryManifestService";
+import { DBTProject } from "../manifest/dbtProject";
+
+// Extends vscode commentthread and add extra fields for reference
+export interface ConversationCommentThread extends CommentThread {
+  share_id: SharedDoc["share_id"];
+  conversation_group_id: ConversationGroup["conversation_group_id"];
+  meta: ConversationGroup["meta"];
+}
+
+export class ConversationComment implements Comment {
+  label: string | undefined;
+  timestamp: Date;
+  savedBody: string | MarkdownString; // for the Cancel button
+  constructor(
+    public conversation_id: Conversation["conversation_id"],
+    public body: string | MarkdownString,
+    public mode: CommentMode,
+    public author: CommentAuthorInformation,
+    public time: string,
+    public parent?: ConversationCommentThread,
+    public contextValue?: string,
+  ) {
+    this.savedBody = this.body;
+    this.timestamp = new Date(this.time);
+  }
+}
+
+const ALLOWED_FILE_EXTENSIONS = [".sql"];
+@provideSingleton(ConversationProvider)
+export class ConversationProvider implements Disposable {
+  private disposables: Disposable[] = [];
+  private _commentController;
```

Comment:
Why `_`, let's not introduce inconsistent variable naming.

Reply:
fixed

Reply:
Not consistently fixed.

Reply:
@mdesmet WDYM?